base/libgit2/utils.jl

```diff
@@ -37,12 +37,16 @@ function prompt(msg::AbstractString; default::AbstractString="", password::Bool=
     end
     msg = !isempty(default) ? msg*" [$default]:" : msg*":"
     uinput = if password
-        Base.getpass(msg)
+        Base.getpass(msg)  # Automatically chomps. We cannot tell EOF from '\n'.
     else
         print(msg)
-        readline()
+        readline(chomp=false)
     end
-    isempty(uinput) ? default : uinput
+    if !password
+        isempty(uinput) && return Nullable{String}()  # Encountered EOF
+        uinput = chomp(uinput)
+    end
+    Nullable{String}(isempty(uinput) ? default : uinput)
```

Comment:
I came up with a better plan. Since this function isn't LibGit2 specific I'll moved this function into `Base`. That way we can have `Base.prompt` and the deprecated `LibGit2.prompt` without having to change the method parameters.